backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/repository/querybuilder/ExpenseCalculatorQueryBuilder.java

```diff
@@ -37,6 +47,107 @@ public String getMusterRollsOfContract(String contractId, String billType, List<
         return queryBuilder.toString();
     }
 
+    public String getBillIds(CalculatorSearchRequest calculatorSearchRequest, List<Object> preparedStmtList) {
+        StringBuilder queryBuilder = new StringBuilder(FETCH_CALCULATE_BILL_IDS_QUERY);
+
```

Comment:
your checking for empty values twice, you dont have to create a new list and add them, you can directly just check for null and construct query